### DISCUSSION TITLE
How to toggle class with alpine ?

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [alpinejs/alpine#1331] [1. Help] [answered]
I want to remove a class and add a new class based on a condition, but when I use x-bind:class it doesn't work.

![image](https://user-images.githubusercontent.com/61914365/119081013-2083b680-b9c9-11eb-8648-b7881d744dcc.png)

Here what I am trying to do is remove 'cursor-not-allowed' class and add 'cursor-pointer' class when `accept` is `true`

**Comment by gwleuverink** [comment, marked as answer]
You can add however many classes & expressions to the object you need:

`:class="{ cursor-pointer': accept, 'cursor-not-allowed' : !accept }"`

**Reply by Chandler-Zhu** [reply]
but when there are multiple conditions it gets a bit messy, such as
`:class="{ 'cursor-pointer': cond_1 && cond_2 && cond_3, 'cursor-not-allowed' : !cond_1 || !cond_2 || !cond_3}"`
is there a better way to implement this ?

**Reply by gwleuverink** [reply]
I'm not aware of any. You could make it easier to parse by moving it to separate lines & extract those conditions to a readable method.

``` html
<div
    :class="{
        'cursor-pointer': hasSomeState(), 
        'cursor-not-allowed': hasSomeOtherState()
    }"
>
```
